nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-security/src/main/java/org/apache/nifi/web/security/jwt/JwtAuthenticationFilter.java

```diff
@@ -43,15 +46,27 @@ public Authentication attemptAuthentication(final HttpServletRequest request) {
         // TODO: Refactor request header extraction logic to shared utility as it is duplicated in AccessResource
 
         // get the principal out of the user token
-        final String authorization = request.getHeader(AUTHORIZATION);
+        final String authentication = request.getHeader(AUTHORIZATION);
 
-        // if there is no authorization header, we don't know the user
-        if (authorization == null || !StringUtils.startsWith(authorization, BEARER)) {
+        // if there is no authorization (authentication) header, we don't know the user
+        if (authentication == null || !validJwtFormat(authentication)) {
             return null;
         } else {
             // Extract the Base64 encoded token from the Authorization header
-            final String token = StringUtils.substringAfterLast(authorization, " ");
+            final String token = getTokenFromHeader(authentication);
             return new JwtAuthenticationRequestToken(token, request.getRemoteAddr());
         }
     }
+
+    private boolean validJwtFormat(String authenticationHeader) {
+        Matcher matcher = tokenPattern.matcher(authenticationHeader);
+        return matcher.matches();
+    }
+
+    private String getTokenFromHeader(String authenticationHeader) {
+        Matcher matcher = tokenPattern.matcher(authenticationHeader);
```

Comment:
I think here we want logic like 
```
if (matcher.find()) { 
    return matcher.group(1); 
} else { 
    // return some value indicating no token; null, constant, or throw exception
}
```

Reply:
Same.

Reply:
This was added but it may not be showing on the main Conversation page. It is lines 68 - 74.